src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/persistence/PublisherEventsPersistor.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.persistence;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.PerformanceAnalyzerApp;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.Action;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.metrics.ExceptionsAndErrors;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.metrics.RcaRuntimeMetrics;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.persistence.actions.PersistedAction;
+import java.time.Instant;
+import java.util.stream.Collectors;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * A listener that persists all actions published by the publisher to rca.sqlite
+ */
+public class PublisherEventsPersistor {
+    public static final String NAME = "publisher_events_persistor";
+    private static final Logger LOG = LogManager.getLogger(PublisherEventsPersistor.class);
+
+    private final Persistable persistable;
+
+    public PublisherEventsPersistor(final Persistable persistable) {
+        this.persistable = persistable;
+    }
+
+    public void persistAction(final Action action) {
+        long timestamp = Instant.now().toEpochMilli();
+        if (persistable != null) {
```

Comment:
Because `persistable` is assigned in the constructor and this is declared final, maybe we can do something like `Objects.requireNonNull` in the constructor and remove the if else here ?

Reply:
Right. Modified.